From ltsi-dev-bounces@lists.linuxfoundation.org Wed Nov  5 10:39:45 2014
From: Chang Rebecca Swee Fun <rebecca.swee.fun.chang@intel.com>
Date: Wed,  5 Nov 2014 09:39:09 +0800
Subject: [LTSI-dev]  [PATCH 04/16] mfd: lpc_ich: Fix ACPI enable bitmask
To: LTSI Mailing List <ltsi-dev@lists.linuxfoundation.org>
Cc: Chang Rebecca Swee Fun <rebecca.swee.fun.chang@intel.com>
Message-ID: <1415151561-16047-5-git-send-email-rebecca.swee.fun.chang@intel.com>


From: Peter Tyser <ptyser@xes-inc.com>

The original bitmask of 0x10 was incorrect and would result in a write
to a reserved read-only bit instead of enabling the ACPI I/O
region.  Update it to the proper value of 0x80.

Signed-off-by: Peter Tyser <ptyser@xes-inc.com>
Tested-by: Rajat Jain <rajatjain@juniper.net>
Reviewed-by: Guenter Roeck <linux@roeck-us.net>
Signed-off-by: Lee Jones <lee.jones@linaro.org>
(cherry picked from commit f5dccb15877b82a40950c6f752d5345c86189fc9)

Signed-off-by: Chang Rebecca Swee Fun <rebecca.swee.fun.chang@intel.com>
---
 drivers/mfd/lpc_ich.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/mfd/lpc_ich.c
+++ b/drivers/mfd/lpc_ich.c
@@ -759,7 +759,7 @@ static void lpc_ich_enable_acpi_space(st
 	u8 reg_save;
 
 	pci_read_config_byte(dev, priv->acpi.ctrl, &reg_save);
-	pci_write_config_byte(dev, priv->acpi.ctrl, reg_save | 0x10);
+	pci_write_config_byte(dev, priv->acpi.ctrl, reg_save | 0x80);
 	priv->acpi.save = reg_save;
 }
 
